### PR TITLE
Filesize limit for logoupload

### PR DESCRIPTION
**Pull request by andrew-secret**
This PR closes https://github.com/owncast/owncast/issues/2480

### Changes made:

- limit file size to 2 MB before API call is fired to the backend
- add helper function to display file size in a proper size unit
- adjust wording below logo upload

# Upload failed with:
image

--

I couldn't upload a real image on localhost, maybe someone can assist me to test it properly. But in theory it should work.

### CODE REVIEW

**Comment by gabek** [comment]
Super weird, you're right, the actual uploading of a logo is flaky. It's been fine previously as far as I know. It seems like sometimes it just doesn't upload any data:

image

I went ahead and added some error handling so it at least returns something and doesn't just break the backend when testing this PR.

File: web/components/config/EditLogo.tsx [hunk 1]
```diff
@@ -43,10 +48,20 @@ export const EditLogo: FC = () => {
 
   // validate file type and create base64 encoded img
   const beforeUpload = (file: RcFile) => {
+    console.log({ file });
```
Comment (gabek) [inline web/components/config/EditLogo.tsx]: Remove this before merging in.

**Comment by gabek** [comment]
Filed #2524 to look at the issue with uploading the actual image.

File: web/components/config/EditLogo.tsx (hunk 1 above)
Comment (andrew-secret) [reply web/components/config/EditLogo.tsx]: Ah good catch. Done ✅

**Comment by gabek** [comment]
Thanks for adding this file size check and messaging around it, that'll certainly clear things up for people in the future.

**Pull request merged**